### PR TITLE
Redmine basic auth. Resolve #295.

### LINKED ISSUE #295: Basic Auth config
I'm trying to use bugwarrior with a redmine that sits behind a Basic Auth. I couldn't find any documentation for how to configure the .bugwarriorrc to handle that. Is this possible?

Thanks!

### PR DESCRIPTION
**Pull request by ryneeverett**
Thanks to @rocketnova for the initial implementation. Note that I do not have a redmine instance to test against.

### CODE REVIEW

File: bugwarrior/services/__init__.py [hunk 1]
```diff
@@ -141,7 +141,7 @@ def config_get(self, key=None, to_type=None):
 
     def config_get_password(self, key, login):
         password = self.config_get_default(key)
-        keyring_service = self.get_keyring_service(self.config, self.target),
+        keyring_service = self.get_keyring_service(self.config, self.target)
```
Comment (ralphbean) [inline bugwarrior/services/__init__.py]: You were correct to remove this comma.  Sorry about that!

**Comment by ralphbean** [comment]
:+1: to replacing ``urllib2`` with ``requests``.  Eager to see if this works for @rocketnova.  :)

**Comment by rocketnova** [comment]
:smile: this totally works for me! Thanks for doing the implementation the right way :+1:

**Comment by ralphbean** [comment]
Let's merge it.  :)

**Pull request merged**